lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -664,13 +665,21 @@ def set_payload(record, entry)
       end
     end
 
-    def log_name(tag)
+    def log_name(tag, commonLabels)
       if @service_name == CLOUDFUNCTIONS_SERVICE
         return 'cloud-functions'
-      else
-        # Add a prefix to VMEngines logs to prevent namespace collisions.
-        return @running_on_managed_vm ? "#{APPENGINE_SERVICE}/#{tag}" : tag
+      elsif @running_on_managed_vm
+        # Add a prefix to Managed VM logs to prevent namespace collisions.
+        return "#{APPENGINE_SERVICE}/#{tag}"
+      elsif @service_name == CONTAINER_SERVICE
+        # For Kubernetes logs, use just the container name as the log name
+        # if we have it.
+        container_name_key = "#{CONTAINER_SERVICE}/container_name"
+        if commonLabels && commonLabels.key?(container_name_key)
+          return commonLabels[container_name_key]
+        end
       end
+      tag
```

Comment:
ha, i was trying to comment on this but i see you already fixed it.  One of the things I disagree with the most, but...